bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/install/IDFDownloadPage.java

```diff
@@ -455,6 +482,95 @@ private void validate()
 		}
 	}
 
+	private String getIdfVersionUsingGit(String idfPath)
+	{
+		if (!validateGitAndPython())
+			return StringUtil.EMPTY;
+		
+		List<String> commands = new ArrayList<String>();
+		commands.add(gitExecutablePath);
+		commands.add("describe"); //$NON-NLS-1$
+		
+		ProcessBuilderFactory processRunner = new ProcessBuilderFactory();
+		try
+		{
+			Logger.log("Executing commads: " + commands.toString());
+			Map<String, String> environment = new HashMap<>(System.getenv());
+			IStatus status = processRunner.runInBackground(commands, Path.fromOSString(idfPath), environment);
+			if (status == null)
+			{
+				Logger.log(IDFCorePlugin.getPlugin(),
+						IDFCorePlugin.errorStatus("Unable to get the process status.", null)); //$NON-NLS-1$
+			}
+			String cmdOutput = status.getMessage();
```

Comment:
we have check for status == null but we are not using break in that condition, so here is possible NPE

Reply:
Updated